contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,497 @@
+---
+oep-number: draft Upgrade 20190710
+title: Upgrade via Kubernetes Job
+authors:
+  - "@kmova"
+owners:
+  - "@amitkumardas"
+  - "@vishnuitta"
+editor: "@kmova"
+creation-date: 2019-07-10
+last-updated: 2019-07-29
+status: implementable
+see-also:
+  - NA
+replaces:
+  - current upgrade steps with scripts
+superseded-by:
+  - NA
+---
+
+# Upgrade via Kubernetes Job
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+    * [Design Constraints](#design-constraints)
+    * [Proposed Implementation](#proposed-implementation)
+    * [High Level Design](#high-level-design)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This design is aimed at providing a design for upgrading the data
+plane components via `kubectl` and also allow administrators to
+automate the upgrade process with higher level operators.
+
+This proposed design will be rolled out in phases while maintaining
+backward compatibility with the upgrade process defined in prior
+releases starting with OpenEBS 1.1. At a high level the design is
+implemented in the following phases:
+- Phase 1: Ability to perform storage pool and volume upgrades using 
+  a Kubernetes Job
+- Phase 2: Allow for saving the history of upgrades on a given pool
+  or volume on a Kubernetes custom resource called `UpgradeTask`
```

Comment:
In phase 2, it will be mainly used for storing the result, but in phase 3, the upgrade operator can watch on a `UpgradeTask` CR and create the corresponding Upgrade job. updated the text.